Fix building against 5.9 or newer kernels

DECLARE_TASKLET was replaced with DECLARE_TASKLET_OLD in 5.9.
Marking as inappropriate for upstream for now, as it seems the
standalone version of the driver has been superceded by the version
in the staging tree in the mainline kernel, and there are no obvious
signs of maintenance.

Upstream-Status: Inappropriate [no upstream]
Signed-off-by: Scott Murray <scott.murray@konsulko.com>

---
 driver/hdm-dim2/dim2_hdm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hdm-dim2/dim2_hdm.c b/hdm-dim2/dim2_hdm.c
index 1847091..fc02c6b 100644
--- a/hdm-dim2/dim2_hdm.c
+++ b/hdm-dim2/dim2_hdm.c
@@ -14,6 +14,7 @@
 #define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
 
 #include <linux/module.h>
+#include <linux/version.h>
 #include <linux/of_platform.h>
 #include <linux/printk.h>
 #include <linux/kernel.h>
@@ -53,7 +54,11 @@ MODULE_PARM_DESC(fcnt, "Num of frames per sub-buffer for sync channels as a powe
 static DEFINE_SPINLOCK(dim_lock);
 
 static void dim2_tasklet_fn(unsigned long data);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,9,0)
 static DECLARE_TASKLET(dim2_tasklet, dim2_tasklet_fn, 0);
+#else
+static DECLARE_TASKLET_OLD(dim2_tasklet, dim2_tasklet_fn);
+#endif
 
 /**
  * struct hdm_channel - private structure to keep channel specific data
